tools/federation/pom.xml

```diff
@@ -0,0 +1,173 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+
+  <modelVersion>4.0.0</modelVersion>
+
+  <artifactId>rdf4j-tools-federation</artifactId>
+  
+  <name>RDF4J: Federation</name>
+  <description>A federation engine for virtually integrating SPARQL endpoints</description>
+  
+  <parent>
+	<groupId>org.eclipse.rdf4j</groupId>
+	<artifactId>rdf4j-tools</artifactId>
+	<version>3.1.0-SNAPSHOT</version>
+  </parent>
+	
+
+  
+  <properties>
+		<jetty.version>9.4.19.v20190610</jetty.version>
+	</properties>
+  
+  
+  <build>
+    <plugins>
+      <plugin>
+          <groupId>org.apache.maven.plugins</groupId>
+          <artifactId>maven-surefire-plugin</artifactId>
+          <version>2.22.1</version>
+          <configuration>
+            <excludes>
+              <exclude>demos/**</exclude>
+              <exclude>local/**</exclude>
+              <exclude>performance/**</exclude>
+            </excludes>
+            <systemPropertyVariables>
+              <repositoryType>SPARQLREPOSITORY</repositoryType>
+            </systemPropertyVariables>
+          </configuration>
+          
+          <executions>
+            <execution>
+             <id>run-tests-native</id>
+             <phase>test</phase>
+             <goals>
+               <goal>test</goal>
+             </goals>
+             <configuration>
+               <systemPropertyVariables>
+                 <repositoryType>NATIVE</repositoryType>
+               </systemPropertyVariables>
+             </configuration>
+           </execution>
+           
+           <execution>
```

Comment:
@jeenbroekstra to save time during the build (or during quickcheck) we can also skip the tests against a HTTP remote repository.
Basically we run the same test set (with a few exceptions) in three environments: SPARQL repository, Native Store, and HTTP repository.
SPARQL and HTTP repository are pretty much similar. In the native store different optimization strategies are applied.
What's your opinion?

Reply:
> @jeenbroekstra I cannot really nail down what's going on with the failing test. It seems to be an OS specific issue (on Mac it is working, on Unix apparently not). The same embeeded test server is also used for the other tests, so it cannot be a general problem.
> 
> I do not want to try & error with commits, can you maybe do a short debugging locally (maybe with more logging enabled in the logback configuration file, see test resources). Hopefully this gives some more hints. Otherwise I need to run the tests in a docker container or a VM.

From what I can tell, for some reason the test expects a remote repository with id `myFederation` to be present on the server, but when I check the test server config, it only contains repositories `endpoint1`, `endpoint2`, `endpoint3`, and `endpoint4`. Does that help at all?

Reply:
I think we can probably be a bit more efficient yeah, though currently (on my machine) the entire fedx test suite runs in about 24 seconds. So I'm not _too_ worried about this :) 

We're more generally, for the entire RDF4J framework, looking into streamlining the tests a bit more: doing more unit tests, and using more mocking and stubbing where possible to avoid spinning up servers all the time.